encoding/test_encoding.py

```diff
@@ -0,0 +1,242 @@
+        )
+        self.assertFalse(got_response, "There should be no response")
+
+
+class TestBothTEAndCE(tester.TempestaTest, CommonUtils):
+    """
+    Ensures that if a response from the backend contains both
+    `Transfer-Encoding: gzip, chunked` and `Content-Encoding: br`,
+    then Tempesta returns 502.
+
+    """
+
+    clients = [
+        {"id": "client", "type": "deproxy", "addr": "${tempesta_ip}", "port": "80"}
+    ]
+    backends = [
+        {
+            "id": "backend",
+            "type": "deproxy",
+            "port": "8000",
+            "response": "static",
+            "response_content":
+            "HTTP/1.1 200 OK\r\n"
+            "Connection: keep-alive\r\n"
+            "Content-type: text/html\r\n"
+            f"Last-Modified: {DATE}\r\n"
+            f"Date: {DATE}\r\n"
+            "Server: Deproxy Server\r\n"
+            "Transfer-Encoding: gzip, chunked\r\n"
+            "Content-Encoding: br\r\n\r\n"
+            "the body does not actually matter",
+        }
+    ]
+    tempesta = {
+        "config": """
+        server ${server_ip}:8000;
+        """
+    }
+
+    def test(self):
+        self.start_all()
+
+        client = self.get_client("client")
+        server = self.get_server("backend")
+
+        got_response = self.send_req(client)
+        response = client.responses[-1] if len(client.responses) else None
+        self.assertTrue(got_response, "There should be a response")
+        self.assertEqual(
+            getattr(response, "status", None), "502", "Wrong response status"
+        )
+
+
+class TestBothTEAndCL(tester.TempestaTest, CommonUtils):
+    """
+    Ensures that if a response from the backend contains both
+    `Transfer-Encoding` and `Content-Length` headers, then
+    Tempesta drops the connection.
+    """
+
+    clients = [
+        {"id": "client", "type": "deproxy", "addr": "${tempesta_ip}", "port": "80"}
+    ]
+    backends = [
+        {
+            "id": "backend",
+            "type": "deproxy",
+            "port": "8000",
+            "response": "static",
+            "response_content":
+            "HTTP/1.1 200 OK\r\n"
+            "Connection: keep-alive\r\n"
+            "Content-type: text/html\r\n"
+            f"Last-Modified: {DATE}\r\n"
+            f"Date: {DATE}\r\n"
+            "Server: Deproxy Server\r\n"
+            "Transfer-Encoding: gzip, chunked\r\n"
+            "Content-Length: 33\r\n\r\n"
+            "the body does not actually matter",
+        }
+    ]
+    tempesta = {
+        "config": """
+        server ${server_ip}:8000;
+        """
+    }
+
+    def test(self):
+        self.start_all()
+
+        client = self.get_client("client")
+        server = self.get_server("backend")
+
+        got_response = self.send_req(client)
+        self.assertTrue(
+            client.connection_is_closed(), "The connection has to be closed"
+        )
+        self.assertFalse(got_response, "There should be no response")
```

Comment:
Responses that have `Transfer-Encoding chunked` and `Content-Length` must be dropped by tempesta and response 502 must be returned to client.

Reply:
It does return a `502`, but it doesn't close the connection for some reason.
```
test (encoding.test_encoding.TestBothTEAndCL) ... HTTP/1.1 502 Bad Gateway
date: Mon, 10 Oct 2022 13:11:53 GMT
content-length: 0
server: Tempesta FW/pre-0.7.0
connection: keep-alive


FAIL

======================================================================
FAIL: test (encoding.test_encoding.TestBothTEAndCL)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/root/tempesta-test/encoding/test_encoding.py", line 181, in test
    self.assertTrue(
AssertionError: False is not true : The connection has to be closed

----------------------------------------------------------------------
```
Even though I manually close it on the backend's side and put a `Connection: close` header into the response, Tempesta for some reason adds `connection: keep-alive` to it. Am I doing something wrong with closing the connections?